cmd/fs-v1_test.go

```diff
@@ -146,19 +143,19 @@ func TestFSPutObject(t *testing.T) {
 	}
 
 	// With a directory object.
-	_, err = obj.PutObject(bucketName+"non-existent", objectName+"/", int64(0), bytes.NewReader([]byte("")), nil, sha256sum)
+	_, err = obj.PutObject(bucketName+"non-existent", objectName+"/", NewHashReader(bytes.NewReader([]byte("abcd")), 0, "", ""), nil)
```

Comment:
bytes.NewReader should be empty?

Reply:
Actually correct but I would still keep the new version because the old test don't really test the 0-size content length. It basically return io.EOF immediately on the first right. The internal LimitReader wouldn't be tested.